search.go

```diff
@@ -0,0 +1,43 @@
+package filter
+
+import (
+	"gorm.io/gorm"
+)
+
+type Search struct {
+	Fields []string
+	Query  string
+}
+
+func (s *Search) Scopes(settings *Settings, modelIdentity *modelIdentity) func(*gorm.DB) *gorm.DB {
+	var fields []string
+
+	// Remove columns that not exist in the table
+	for _, field := range s.Fields {
+		_, ok := modelIdentity.Columns[field]
+		if ok {
+			fields = append(fields, field)
+		}
+	}
+
+	if len(fields) == 0 {
+		return nil
+	}
+
+	return func(tx *gorm.DB) *gorm.DB {
+		searchQuery := tx.Session(&gorm.Session{})
+
+		for _, field := range fields {
+			filter := &Filter{
+				Field:    field,
+				Operator: settings.SearchOperator,
```

Comment:
Use operator `$cont` by default if `settings.SearchOperator` is `nil` (not defined in `Settings`).